sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormat.scala

```diff
@@ -265,8 +265,10 @@ object FileFormat {
    * fields of the [[PartitionedFile]], and do have entries in the file's metadata map.
    */
   val BASE_METADATA_EXTRACTORS: Map[String, PartitionedFile => Any] = Map(
-    FILE_PATH -> { pf: PartitionedFile => pf.toPath.toString },
-    FILE_NAME -> { pf: PartitionedFile => pf.toPath.getName },
+    FILE_PATH -> { pf: PartitionedFile => pf.filePath.urlEncoded },
```

Comment:
what's the behavior in Spark 3.3? Does Spark normalize the path string regarding `file:/...` and `file:///...`?

Reply:
I tried this
```
scala> sql("create table p1 using parquet location 'file:/tmp/path'")
res2: org.apache.spark.sql.DataFrame = []

scala> sql("create table p2 using parquet location 'file:///tmp/path'")
res3: org.apache.spark.sql.DataFrame = []

scala> sql("select _metadata.file_path from p1").show
+--------------------+
|           file_path|
+--------------------+
|file:/tmp/path/pa...|
|file:/tmp/path/pa...|
|file:/tmp/path/pa...|
|file:/tmp/path/pa...|
|file:/tmp/path/pa...|
|file:/tmp/path/pa...|
|file:/tmp/path/pa...|
|file:/tmp/path/pa...|
|file:/tmp/path/pa...|
|file:/tmp/path/pa...|
+--------------------+


scala> sql("select _metadata.file_path from p2").show
+--------------------+
|           file_path|
+--------------------+
|file:/tmp/path/pa...|
|file:/tmp/path/pa...|
|file:/tmp/path/pa...|
|file:/tmp/path/pa...|
|file:/tmp/path/pa...|
|file:/tmp/path/pa...|
|file:/tmp/path/pa...|
|file:/tmp/path/pa...|
|file:/tmp/path/pa...|
|file:/tmp/path/pa...|
+--------------------+
```

To avoid regression I think we should do normalization here as well.